redaxo/src/core/lib/sql/sql.php

```diff
@@ -1463,4 +1502,53 @@ public static function checkDbConnection($host, $login, $pw, $dbname, $createDb
 
         return  $err_msg;
     }
+
+    private function setMultiRecordQuery($verb)
+    {
+        $fields = [];
+
+        foreach ($this->records as $record) {
+            foreach ($record->values as $field => $value) {
+                $fields[$field] = true;
+            }
+            foreach ($record->rawValues as $field => $value) {
+                $fields[$field] = true;
+            }
+        }
+
+        $fields = array_keys($fields);
+
+        $rows = [];
+        $params = [];
+
+        foreach ($this->records as $record) {
+            $row = [];
+
+            foreach ($fields as $field) {
+                if (isset($record->rawValues[$field])) {
+                    $row[] = $record->rawValues[$field];
+
+                    continue;
+                }
+
+                if (!isset($record->values[$field]) && !array_key_exists($field, $this->values)) {
+                    $row[] = 'DEFAULT';
+
+                    continue;
+                }
+
+                $row[] = '?';
+                $params[] = $record->values[$field];
```

Comment:
Ist es hier überhaupt sinnvoll mit prepared statements zu arbeiten?
Können unter Umständen ja sehr viele Parameter werden, da gibt es doch bestimmt auch Limits, oder?
Daher überlege ich, ob man hier besser selbst escapen sollte und die Values direkt in die Query schreiben sollte?

Reply:
Wird meine ich unter anderem durch [`max_allowed_packet`](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet) beeinflusst.
Da müsste man aber beachten, dass bei MySQL 5.7 4MB gesetzt sind, aber bei 5.5 nur 1MB per Default gesetzt sind.

Reply:
Wg anzahl parameter ist mir kein limit bekannt, ich nutze das aber auch selten

Reply:
Da wir die Anzahl ja schon begrenzen durch die Chunks (siehe oben), würde ich es dann erstmal bei prepared Statements lassen? Und dann sammeln wir damit erst Mal Erfahrungen, und passen es ggf. später an, wenn es sich so als nachteilig erweist.